File: Python/simple_linear_regression.py

#Regrsión lineal simple 

# Importar librerias
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importar dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Dividir dataset en entrenamiento y testing
from sklearn.model_selection import train_test_split
X_train, X_test,y_train,y_test = train_test_split(X,y, test_size = 1/3, random_state=0)

# Creacion de modelo de regresion lineal para entrenamiento 
from sklearn.linear_model import LinearRegression
regression = LinearRegression()
regression.fit(X_train, y_train) #es necesario que sean del mismo tamaño para que funcione

# Prediccion de conjunto de test
y_pred =regression.predict(X_test)

#Visualizar resultados de entrenamiento 
plt.scatter(X_train, y_train, color = "red")
plt.plot(X_train, regression.predict(X_train), color = "blue")
plt.title("Sueldo Vs Años de Experiencia(Conjunto de Entrenamiento)")
plt.xlabel("Años de experiencia")
plt.ylabel("Sueldo en dlls")
plt.show()


#Visualizar resultados de test 
plt.scatter(X_test, y_test, color = "red")
#queda igual debido a que la recta de gregresion lineal es unica y ya habiamos hecho el modelado con X_train
plt.plot(X_train, regression.predict(X_train), color = "blue") 
plt.title("Sueldo Vs Años de Experiencia(Conjunto de Entrenamiento)")
plt.xlabel("Años de experiencia")
plt.ylabel("Sueldo en dlls")
plt.show()

File: R/simple_linear_regression.R
#Regresion lineal simple

#Importar dataset
dataset = read.csv('Salary_Data.csv')

#Cargar libreria y dividir conjunto entrenamiento y training
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
testing_set = subset(dataset, split == FALSE)
#Utilizando summary(Regressor) podemos ver muchos datos de nuestra regresion lineal 
#que tan buena es, lo mas importante son los coeficientes la relevancia estadistica
#se representa en los coeficientes con asteriscos *** el maximo de asteriscos es tres
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

#predecir resultados con el congunto de test
#Aqui usamos el modelo de prediccion el cual es regresor y lo comparamos con el testis_set


#Visualizacion de los datos en el conjunto de entrenamiento
#install.packages("ggplot2")
library(ggplot2)
ggplot()+
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
                 colour = "red")+
  geom_line(aes(x =training_set$YearsExperience,
                             y = predict(regressor, newdata = training_set)),
                         colour = "blue")+
  ggtitle("Sueldo vs Años de experiencia ") +
  xlab("Años de experiencia")+
  ylab("Sueldo")

#Visualisacion de los resultados de testing
ggplot()+
  geom_point(aes(x = testing_set$YearsExperience, y = testing_set$Salary),
             colour = "red")+
  geom_line(aes(x =training_set$YearsExperience,
                y = predict(regressor, newdata = training_set)),
            colour = "blue")+
  ggtitle("Sueldo vs Años de experiencia (Testing)") +
  xlab("Años de experiencia")+
  ylab("Sueldo")


